app/backend-dotnet/Program.cs

```diff
@@ -0,0 +1,72 @@
+using Azure;
+using Azure.AI.OpenAI;
+using Azure.Core;
+using Azure.Identity;
+using Azure.Search.Documents;
+using Azure.Storage.Blobs;
+using IMJohn.Services;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Configuration;
+
+var builder = WebApplication.CreateBuilder(args);
+
+// Add services to the container.
+
+builder.Services.AddControllers();
+// Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddSwaggerGen();
+
+var azureCredential = new DefaultAzureCredential();
+var AZURE_STORAGE_ACCOUNT = builder.Configuration.GetValue<string>("AZURE_STORAGE_ACCOUNT");
+
+// Add blob service client
+var blobServiceClient = new BlobServiceClient(new Uri($"https://{AZURE_STORAGE_ACCOUNT}.blob.core.windows.net"), azureCredential);
+builder.Services.Add(ServiceDescriptor.Singleton(blobServiceClient));
```

Comment:
```suggestion
builder.Services.AddSingleton<BlobServiceClient>(provider =>
{
    var config = provider.GetRequiredServices<IConfiguration>();
    var storageAccount = config.GetValue<string>("AZURE_STORAGE_ACCOUNT");
    var azureCredential = new DefaultAzureCredential();

    return new BlobServiceClient(
        new Uri($"https://{storageAccount}.blob.core.windows.net"), azureCredential);
});
```

I'd probably rewrite this as follows.